src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.AssemblyResolution.targets

```diff
@@ -157,4 +157,23 @@ _ResolveAssemblies MSBuild target.
     </ItemGroup>
   </Target>
 
+  <Target Name="_IncludeNativeSystemLibraries">
+    <ItemGroup>
+      <FrameworkNativeLibrary
+          Include="@(ResolvedFileToPublish)"
+          Condition=" '$(AndroidIncludeDebugSymbols)' == 'true' and '%(Filename)%(Extension)' == 'libmono-android.debug.so' "
+          ArchiveFileName="libmonodroid.so"
+      />
+      <FrameworkNativeLibrary
+          Include="@(ResolvedFileToPublish)"
+          Condition=" '$(AndroidIncludeDebugSymbols)' != 'true' and '%(Filename)%(Extension)' == 'libmono-android.release.so' "
+          ArchiveFileName="libmonodroid.so"
+      />
+      <FrameworkNativeLibrary
+          Include="@(ResolvedFileToPublish)"
+          Condition=" '%(Extension)' == '.so' and '%(Filename)' != 'libmono-android.debug' and '%(Filename)' != 'libmono-android.release' "
+      />
+    </ItemGroup>
```

Comment:
How are we going to handle `_AndroidNativeLibraryForFastDev` in the .net 6 world? This is used when we fast deploy most native libraries to the override directory rather than include them in the apk.

Reply:
Hopefully some test will fail here, as this was removed for .NET 6:
```xml
<_AndroidNativeLibraryForFastDev Condition=" '$(_InstantRunEnabled)' == 'True' And '$(AndroidUseDebugRuntime)' == 'True' And '$(_AndroidCheckedBuild)' == '' " Include="%(_TargetLibDir.Identity)\libxamarin-debug-app-helper.so" />
```
I'll need to add an equivalent for these.

Reply:
`libxamarin-debug-app-helper.so` is in our runtime packs and will end up as a `@(FrameworkNativeLibrary)`, and those items aren't fast deployed. Before https://github.com/xamarin/xamarin-android/commit/fb727f0a64ac2932b96a117c4a26786eac8e0f6c, these files were hard coded in the `<BuildApk/>` task.

It looks like only  `@(AndroidNativeLibrary)` items are added to `@(_AndroidNativeLibraryForFastDev)`: https://github.com/xamarin/monodroid/blob/27736a7ffc48d606ab45598f761e873f8572f46a/tools/msbuild/Xamarin.Android.Common.Debugging.targets#L224. 

Maybe we should look at fast deploying `@(FrameworkNativeLibrary)` in a future PR? If we try to fast deploy both customer libs and framework libs we'll need to rework: https://github.com/xamarin/xamarin-android/blob/aaa37c3df94490ee9befb8381f4dd7aa18226355/src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs#L589

As for the three `$(_AndroidNativeLibraryForFastDev)` items explicitly mentioned here, two files are used for native static code analysis (which I don't think we ever need to put into customer apps).  The other one is for the interpreter, and I am not sure if that is available in .NET 6? I am not sure if we need to worry about any of these in a .NET 6 context?

Reply:
certain native libs still need to be in the apk even when fast deploying. specifically monodroid.so and its immediate dependencies. This is because it is loaded as part of the runtime startup. So we can't just fast dev everything.

Reply:
The latest changes shouldn't change the current logic for .NET 6, except the block:

```xml
<!-- HACK remove any unexpected runtime components -->
<FrameworkNativeLibrary
    Remove="@(ResolvedFileToPublish)"
    Condition="  $([System.Text.RegularExpressions.Regex]::IsMatch('%(FullPath)', '.*linux-.*'))  "
/>
```

Instead of silently ignoring, you get a `XA4301` warning now.

The tests we have for Fast Deployment should be passing, so I don't think there is actually anything broken?